code/game/machinery/magnet.dm

```diff
@@ -361,13 +360,12 @@
 		pathpos++ // increase iterator
 
 		// Broadcast the signal
-		spawn()
-			radio_connection.post_signal(src, signal, filter = RADIO_MAGNETS)
+		INVOKE_ASYNC(radio_connection, /datum/radio_frequency.proc/post_signal, src, signal, RADIO_MAGNETS)
 
 		if(speed == 10)
-			sleep(1)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 1)
 		else
-			sleep(12-speed)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 12-speed)

```

Comment:
OK, if you actually read the code, you'd understand why this doesn't preserve the old behaviour at all